src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4102,40 +4130,56 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	 * aggregates, we can push down the original limit. Else if we have order by
 	 * clauses with commutative aggregates, we can push down approximate limits.
 	 */
-	if (groupClauseList == NIL)
+	if (groupClauseList == NIL || sortClauseList == NIL || !hasOrderByAggregate)
```

Comment:
Yeah, makes sense. I have added a check for queries with aggregates but no `ORDER BY`. Do you have any other conditions that we should not push `LIMIT` down?